python/pyiceberg/schema.py

```diff
@@ -1367,3 +1382,258 @@ def _(file_type: DecimalType, read_type: IcebergType) -> IcebergType:
             raise ResolveError(f"Cannot reduce precision from {file_type} to {read_type}")
     else:
         raise ResolveError(f"Cannot promote an decimal to {read_type}")
+
+
+class UpdateSchema(ABC):
```

Comment:
Do we need this ABC? Feels a bit Java'ish, also the naming is confusing `UpdateSchema` and `SchemaUpdate` as the implementation

Reply:
I moved `SchemaUpdate`  to `table/__init__.py` as an internal private class, and do you think that's better?

The current names reference the java code names, so what do we change them to?

Reply:
> Do we need this ABC? Feels a bit Java'ish

I dropped that `ABC`. Does keeping `ABC` make it clearer that it is an interface?

Reply:
removed.